src/internal/utils/mdObjectHelper.ts

```diff
@@ -59,6 +59,27 @@ export function hasMeasures(mdObject: VisualizationObject.IVisualizationObjectCo
     return mdObject && getMeasuresFromMdObject(mdObject).length > 0;
 }
 
+export function hasSizeMeasure(mdObject: VisualizationObject.IVisualizationObjectContent): boolean {
+    const sizeBucket = mdObject && findBucketByLocalIdentifier(mdObject.buckets, BucketNames.SIZE);
+    return sizeBucket && get(sizeBucket, "items").length > 0;
+}
+
+export function hasLocation(mdObject: VisualizationObject.IVisualizationObjectContent): boolean {
+    if (!mdObject) {
+        return false;
+    }
+
+    return getBucketItems(mdObject.buckets, BucketNames.LOCATION).length > 0;
+}
+
+export function hasSegment(mdObject: VisualizationObject.IVisualizationObjectContent): boolean {
```

Comment:
Since we have `hasSizeMeasure`, so do `hasLocationAttribute` and `hasSegmentAttribute`